op-node/rollup/derive/span_channel_out.go

```diff
@@ -166,20 +194,24 @@ func (co *SpanChannelOut) AddSingularBatch(batch *SingularBatch, seqNum uint64)
 	if err = co.compress(); err != nil {
 		return err
 	}
-	co.lastCompressedRLPSize = co.activeRLP().Len()
 
 	// if the channel is now full, either return the compressed data, or the compressed previous data
 	if err := co.FullErr(); err != nil {
-		// if there is only one batch in the channel, it *must* be returned
-		if len(co.spanBatch.Batches) == 1 {
+		// if it's the first batch of the channel, it *must* fit in
+		if co.sealedRLPBytes == 0 && co.spanBatch.GetBlockCount() == 1 {
+			return nil
+		}
+
+		// if we just perfectly filled up the channel, also return nil to retain block
+		if uint64(co.compressor.Len()) == co.target {
 			return nil
 		}
```

Comment:
nice catch :)

For other reviewers - returning nil here is a good idea because there's nothing to roll back. On the *next* attempted addition, or on any fullness check, the Channel Out will report full.